### DISCUSSION TITLE
Body Battery Data Field

### DISCUSSION THREAD

**Discussion by gmardau** [SylvainGa/crystal-face#3] [General]
Hey Sylvain. First of all, thank you so very much for publishing this app. Was walking around with my new Fenix 7 for a few months now without a proper watch face on it.

My request would be to add the Body Battery option to the data fields, if that is possible to retrieve.

Thanks.

**Comment by SylvainGa** [comment]
Moved to Discussion

I'll look into it but from a quick search, I do not see it has a sensor that can be queried,

https://developer.garmin.com/connect-iq/api-docs/Toybox/Sensor/Info.html

Edit: And in the developer forum, a post from last year says it's not available, neither is the stress level.

**Comment by k0d3r1s** [comment]
it is bit hidden but there:  
https://developer.garmin.com/connect-iq/api-docs/Toybox/SensorHistory.html#getBodyBatteryHistory-instance_function  
https://forums.garmin.com/developer/connect-iq/b/news-announcements/posts/connect-iq-system-5-sdk-now-available

**Comment by SylvainGa** [comment]
Ok thanks, I'll see if I can use it for the purpose of displaying current data.

**Comment by SylvainGa** [comment]
V2.10.0 now has Body Battery. Should appear soon on ConnectIQ

**Reply by gmardau** [reply]
Many, many thanks Sylvain, it's working great!!

**Comment by SylvainGa** [comment]
#response_container_BBPPID{font-family: initial; font-size:initial; color: initial;}  Thanks for the feedback.                                                                                                                                                                                                                                                                                                                        De : ***@***.***é : 22 juillet 2022 05 h 52À : ***@***.***épondre à : ***@***.*** : ***@***.***; ***@***.*** : Re: [SylvainGa/crystal-face] Body Battery Data Field (Discussion #3)  
Many, many thanks Sylvain, it's working great!!

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

**Comment by maximkuzmich** [comment]
@SylvainGa , at first, thank you very much for getting Crystal watch face back to live :) 

I have a question about the Body Batter metric. Can this metric be added to the left/right meter? It could be helpful to see your own "battery" draining more visually (the watch battery can be configured to be shown there).

**Reply by SylvainGa** [reply]
Hi, version 2.13.0 that I just uploaded has Body Battery as a Goal Meter.
